aiohttp/connector.py

```diff
@@ -770,6 +774,13 @@ def __init__(
         self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
         self._family = family
         self._local_addr = local_addr
+        self._network_interface = network_interface
+        if self._local_addr and self._network_interface:
+            raise ValueError(
+                "local_addr and network_interface can't be defined at the same time."
```

Comment:
Why are these incompatible? Couldn't you just pass the local_addr arguments to .connect_ex() (line 986)?

Reply:
No, local_addr is the source ip. You don't wanna connect to yourself.

Reply:
However, i agree the user should have full control and for future compatibility I will remove the check and allow binding.

Reply:
I have allowed support for custom source ip on custom nic and removed the check.